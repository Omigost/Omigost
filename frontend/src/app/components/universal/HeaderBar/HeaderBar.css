@import "../../../css/Theme.less";

.HeaderBar {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: @HeaderBarHeight; 
  border-bottom: 1px solid transparent;
  box-shadow: 0px 2px 10px -7px rgba(0,0,0,0.5);
  background-color: black;
  display: flex;
  z-index: 9999999;
  overflow: hidden;
  transition: all @HeaderAnimationTime;
}

.Hidden {
  height: 0;
}

.HeaderBarCollapsed {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: @HeaderBarHeight / 2; 
  border-bottom: 0.3vw solid #ff471f;
  box-shadow: 0px 2px 10px -7px rgba(0,0,0,0.5);
  background-color: white;
  display: flex;
  z-index: 99999999999;
  transition: all @HeaderAnimationTime;
}

.ExpandableMenuWrapper {
  display: none;
}

.LogoWrapper {
  width: auto;
  height: 100%;
  display: inline-block;
}

.MenuWrapper {
  width: 100%;
  display: inline-block;
  margin-left: @HeaderBarHeight / 7;
}

.LoginViewWrapper {
  position: relative;
  top: 1vw;
  margin-right: 2vw;
  transition: all @HeaderAnimationTime;
}

.HeaderBarCollapsed .LoginViewWrapper {
  position: relative;
  top: -0.1vw;
}

@media only screen and (max-width: 550px) {
  .HeaderBarCollapsed {
    height: 12vh !important;
  }

  .HeaderBarCollapsed .LogoWrapper div {
    height: 100% !important;
  }

  .HeaderBarCollapsed .LogoWrapper img {
    margin-top: 3vh !important;
    height: 6vh !important;
  }

  .HeaderBarCollapsed .MenuWrapper {
    display: none !important;
  }

  .HeaderBarCollapsed .LoginViewWrapper {
    display: none !important;
  }

  .HeaderBarCollapsed .ExpandableMenuWrapper {
    display: inline-block !important;
  }
}