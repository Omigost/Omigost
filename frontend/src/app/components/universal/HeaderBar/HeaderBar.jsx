import React from 'react'
import * as styles from './HeaderBar.css'

import HeaderMenu from 'components/HeaderMenu/HeaderMenu.jsx'
import Logo from 'components/Logo/Logo.jsx'
import TopExpandableMenu from 'components/TopExpandableMenu/TopExpandableMenu.jsx'

class HeaderBar extends React.Component {
  
  constructor(props) {
    super(props);
    this.state = {
      collapsed: false
    };
    this.mounted = false;
    
    this._collapse = this._collapse.bind(this);
    this._scrollListener = this._scrollListener.bind(this);
  }

  _scrollListener(e) {
    const scrollY = window.scrollY || 0;
    if(scrollY > 100) {
      this._collapse(true);
    } else {
      this._collapse(false);
    }
  }
  
  _collapse(shouldCollapse) {
    if(!this.mounted) return;
    this.setState({
      collapsed: shouldCollapse
    });
  }
  
  componentDidMount() {
    this.mounted = true;
    window.addEventListener('scroll', this._scrollListener);
  }
  
  componentWillUnmount() {
    this.mounted = false;
    window.removeEventListener('scroll', this._scrollListener);
  }
  
  render() {
    const isCollapsed = this.state.collapsed;

    //if(!isCollapsed) return null;

    return (
      <div
        styleName={
          (isCollapsed)?('HeaderBarCollapsed'):('HeaderBar')
        }
        className={(!isCollapsed)?(styles.Hidden):('')}
      >
        {
            (isCollapsed)?(
                <div styleName='ExpandableMenuWrapper'>
                    <TopExpandableMenu />
                </div>
            ):(null)
        }
        <div styleName='LogoWrapper'>
          <Logo small />
        </div>
        <div styleName='MenuWrapper'>
          <HeaderMenu isCollapsed={isCollapsed} small plainLoginButton={!isCollapsed}/>
        </div>
      </div>
    )
  }
}

export default HeaderBar