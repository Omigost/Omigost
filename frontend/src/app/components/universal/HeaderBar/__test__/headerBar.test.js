import React from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';

import HeaderBar from '../HeaderBar.jsx';

test('Snapshot test of Header', () => {
    const component = shallow(
        <HeaderBar />
    );

    expect(toJson(component, {
        mode: 'shallow'
    })).toMatchSnapshot();
});