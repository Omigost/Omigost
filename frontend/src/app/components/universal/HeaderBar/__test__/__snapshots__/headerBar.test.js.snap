// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of Header 1`] = `
<div
  styleName="HeaderBar"
>
  <div
    styleName="LogoWrapper"
  >
    <Logo
      small={true}
    />
  </div>
  <div
    styleName="MenuWrapper"
  >
    <PropsContentProvider
      isCollapsed={false}
      plainLoginButton={true}
      small={true}
    />
  </div>
</div>
`;
